opm/simulators/flow/FIBlackoilModel.hpp

```diff
@@ -55,27 +59,69 @@ class FIBlackOilModel : public BlackOilModel<TypeTag>
         historySize = getPropValue<TypeTag, Properties::TimeDiscHistorySize>(),
     };
 
+    // The chunked and threaded iteration over elements in this class assumes that the number
+    // and order of elements is fixed, and is therefore constrained to only work with CpGrid.
+    // For example, ALUGrid supports refinement and can not assume that.
+    static constexpr bool gridIsUnchanging = std::is_same_v<GetPropType<TypeTag, Properties::Grid>, Dune::CpGrid>;
```

Comment:
Very, very strictly speaking you should `¤include <type_traits>` for `std::is_same_v<>`.  It's likely, though, that we get this header transitively, so I'm not going to insist that we do that now.  On a tangential note, we're really missing `<cstddef>` for `std::size_t` too.